### PR TITLE
Android platform view static thread merging

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

Make the raster task runner always run on the platform thread when user enables 'io.flutter.embedded_views_preview` in Manifest.xml

## Related Issues

https://github.com/flutter/flutter/issues/60016

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java [hunk 1]
```diff
@@ -234,13 +233,18 @@ public void ensureInitializationComplete(
...
+
       FlutterJNI.nativeInit(
           applicationContext,
           shellArgs.toArray(new String[0]),
           kernelPath,
           result.appStoragePath,
           result.engineCachesPath,
-          initTimeMillis);
+          initTimeMillis,
+          use_embedded_view);
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java:247]: this also needs a change in FlutterJNI#nativeInit signature

File: shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java [hunk 2]
```diff
@@ -234,13 +233,20 @@ public void ensureInitializationComplete(
...
       long initTimeMillis = SystemClock.uptimeMillis() - initStartTimestampMillis;
+
+      // TODO(cyanlaz): Remove this when dynamic thread merging is done.
+      // https://github.com/flutter/flutter/issues/59930
+      Bundle bundle = applicationInfo.metaData;
+      boolean use_embedded_view = bundle.getBoolean("io.flutter.embedded_views_preview");
+
       FlutterJNI.nativeInit(
           applicationContext,
           shellArgs.toArray(new String[0]),
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java:244]: nit: what about passing the flag in `shellArgs`?

File: common/settings.h [hunk 3]
```diff
@@ -213,6 +213,13 @@ struct Settings {
   /// to log a timeline event that tracks the latency of engine startup.
   std::chrono::microseconds engine_start_timestamp = {};
 
+  /// Does the application claim that it uses the android embedded view for platform views.
```
Comment (blasten) [inline common/settings.h:216]: nit: is this missing a question mark?

File: shell/platform/android/android_shell_holder.cc [hunk 4]
```diff
@@ -101,28 +103,55 @@ AndroidShellHolder::AndroidShellHolder(
...
+    // The plan is to eventually dynamically merge the threads when there's a
+    // platform view in the layer tree.
+    // For now we use a fixed thread configuration with the same thread used as the
+    // gpu and platform task runner.
+    // TODO(amirh/chinmaygarde): remove this, and dynamically change the thread configuration.
+    // https://github.com/flutter/flutter/issues/23975
+    // https://github.com/flutter/flutter/issues/59930
+    flutter::TaskRunners task_runners(thread_label,     // label
+                                      platform_runner,  // platform
+                                      platform_runner,       // raster
```
Comment (blasten) [inline shell/platform/android/android_shell_holder.cc:118]: nit: formatting

**Comment by blasten** [comment]
Until https://github.com/flutter/flutter/issues/55326 is fixed, tests will be added to the framework where we can run e2e tests.

File: common/settings.h (hunk 3 above)
Comment (cyanglaz) [reply common/settings.h]: Updated to "Whether"

File: shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java (hunk 2 above)
Comment (cyanglaz) [reply shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java:247]: done

File: shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java [hunk 5]
```diff
@@ -237,9 +237,11 @@ public void ensureInitializationComplete(
       // TODO(cyanlaz): Remove this when dynamic thread merging is done.
       // https://github.com/flutter/flutter/issues/59930
       Bundle bundle = applicationInfo.metaData;
-      boolean use_embedded_view = bundle.getBoolean("io.flutter.embedded_views_preview");
-      if (use_embedded_view) {
-        shellArgs.add("--use-embedded-view");
+      if (bundle != null) {
+        boolean use_embedded_view = bundle.getBoolean("io.flutter.embedded_views_preview");
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java:241]: minor nit: `io.flutter.embeddedViewsPreview`

Comment (cyanglaz) [reply shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java:241]: offline discussion: We will keep the flag name same as iOS.

**Pull request merged**